tests/scene/test_occluder_instance_3d.h

```diff
@@ -0,0 +1,127 @@
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_OCCLUDER_INSTANCE_3D_H
+#define TEST_OCCLUDER_INSTANCE_3D_H
+
+#include "core/math/vector3.h"
+#include "scene/3d/mesh_instance_3d.h"
+#include "scene/3d/occluder_instance_3d.h"
+#include "scene/main/scene_tree.h"
+#include "scene/resources/3d/primitive_meshes.h"
+
+#include "tests/test_macros.h"
+
+namespace TestOccluderInstance3D {
+
+TEST_CASE("[SceneTree][OccluderInstance3D] Test baking functionality") {
+// We need to load an occluder path because when the light occluder generates
+// it has to save to a file.
+#ifdef WINDOWS_ENABLED
+	const String occluder_path = OS::get_singleton()->get_environment("TEMP").path_join("test_occluder.occ");
+#else
+	const String occluder_path = "/tmp/test_occluder.occ";
+#endif
+
+	SceneTree *scene_tree = SceneTree::get_singleton();
+
+	// Create our occluder to do tests with.
+	OccluderInstance3D *occluder_instance = memnew(OccluderInstance3D);
+
+	// Root node to put meshes under.
+	Node *test_bake_scene = memnew(Node);
+
+	scene_tree->get_root()->add_child(test_bake_scene);
+	scene_tree->get_root()->add_child(occluder_instance);
+
+	// Instantiate a mesh to generate.
+	MeshInstance3D *box_mesh_instance = memnew(MeshInstance3D);
+	Ref<BoxMesh> box_mesh = Ref<BoxMesh>();
+	box_mesh.instantiate();
+	box_mesh_instance->set_mesh(box_mesh);
+	test_bake_scene->add_child(box_mesh_instance);
+	box_mesh_instance->set_owner(test_bake_scene);
+
+	// This is the vertices output that we expect for a single cube.
+	const PackedVector3Array expected_vertices_output = PackedVector3Array(
+			{ Vector3(-0.5, 0.5, 0.5), Vector3(0.5, 0.5, 0.5), Vector3(-0.5, -0.5, 0.5), Vector3(0.5, -0.5, 0.5), Vector3(0.5, 0.5, -0.5), Vector3(-0.5, 0.5, -0.5),
+					Vector3(0.5, -0.5, -0.5), Vector3(-0.5, -0.5, -0.5) });
+
+	// This is the indices output that we expect for a single cube.
+	const PackedInt32Array expected_indices_output = PackedInt32Array(
+			{ 0, 1, 2, 1, 3, 2, 4, 5, 6, 5, 7, 6, 1, 4, 3, 4, 6, 3, 5, 0, 7, 0, 2, 7, 1, 0, 4, 0, 5, 4, 2, 3, 7, 3, 6, 7 });
+
+	// Turn errors off to suppress a warning about occlusion culling disabled at build-time.
```

Comment:
Technically, the [warning message](https://github.com/godotengine/godot/blob/4248411baf7a81b49e827f3f17d9dcc4b53b6c6a/servers/rendering/renderer_scene_occlusion_cull.h#L203-L205) is invalid as we do have occlusion culling enabled at build-time, but it does nothing in headless mode. This could be fixed in a separate PR.